### ISSUE TITLE
Feature to allow map bearing to follow user

### ISSUE THREAD

**Issue by lucasgray** [nitaliano/react-native-mapbox-gl#834]
I'd like to display a map where the rotation of the map is locked to the compass activity from the user's device, so as the user changes their location and rotation, the map is updated to account for this automatically.  

I've looked around a bit and this seems to be available with native android but I've tried all the available User Tracking Modes in this library and it doesn't seem to allow what I'm looking for.  

Is this a feature that can be exposed? Or is it already, and I can't find it?

Thanks!

**Comment by EricPKerr** [comment]
I don't believe this is in the library automatically right now.  You could achieve this by monitoring the user's last few GPS readings and calculating the corresponding headings from those readings (weighted average most likely) and then passing that into `<MapboxGL.MapView heading={...} />`. 

This library can help with the bearing/heading calculation: https://github.com/Turfjs/turf/tree/master/packages/turf-bearing

**Comment by nitaliano** [comment]
@lucasgray closing this out in favor of https://github.com/mapbox/react-native-mapbox-gl/issues/881. It's going to be supported, iOS already has support for this and Android just needs to polyfill it

**Comment by butchmarshall** [comment]
Yep, this was previously working in v5 with the property Mapbox.userTrackingMode.followWithCourse, glad to hear it's going to make it into v6.

**Comment by lucasgray** [comment]
@nitaliano so does that mean @butchmarshall's `followWithCourse` will be the accepted solution to this once the work is complete?

@EricPKerr thanks for the suggestion, it works okay, is a little jittery, I think an out of the box solution would be wonderful.  Thanks all.

**Comment by nitaliano** [comment]
@lucasgray yes, we are going to be porting over what the iOS SDK does for location to Android, these will be the possible values, it says what each one does https://www.mapbox.com/ios-sdk/api/3.7.1/Enums/MGLUserTrackingMode.html. `followWithHeading` will be the accepted solution for this.

**Comment by nitaliano** [comment]
Support for this is going to be added to this PR https://github.com/mapbox/react-native-mapbox-gl/pull/897 so it will land in time for our next npm release `6.0.3`

**Comment by nitaliano** [comment]
PR has been merged into master you can try it out by using `followWithCourse` or `followWithHeading`
